### PR TITLE
dnsdist: Hopefully make the "passing the source address" documentation better

### PR DESCRIPTION
**Pull request by rgacogne**
### Checklist

I have:
- [x] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [ ] compiled this code
- [ ] tested this code
- [x] included documentation (including possible behaviour changes)
- [ ] documented the code
- [ ] added or modified regression test(s)
- [ ] added or modified unit test(s)

### CODE REVIEW

**Comment by rgacogne** [comment]
> typing-extensions==4.12.0 not found

what?

**Comment by rgacogne** [comment]
I see duplicated entries in the auth's doc python dependencies base64 file:
```
typing_extensions==4.11.0
typing_extensions==4.12.0
```

**Comment by rgacogne** [comment]
@romeroalx Any idea what's going on here, by any chance?

**Comment by romeroalx** [comment]
> @romeroalx Any idea what's going on here, by any chance?

Yes, our package `sphinxcontrib-openapi` has a `setup_requires` dependency for `setuptools-scm` which in turn requires a not fixed version of `typing-extensions`.

We need to re-generate the pinned list, but forcing a higher version of that package

File: pdns/dnsdistdist/docs/advanced/passing-source-address.rst [hunk 1]
```diff
@@ -4,19 +4,94 @@ Passing the source address to the backend
...
+--------------
+
+.. note:
+  The Proxy Protocol has been designed by the HAProxy folks for HTTP over TCP, but is generic enough to be used in other places, and is a de-facto standard with implementations in ISC Bind, Knot, Knot Resolver, PowerDNS Authoritative, PowerDNS Recursor, Unbound, HAProxy, nginx, postfix and many others.
+  It works by pre-pending a small header at the very beginning of a UDP datagram or TCP connection, which holds the initial source and destination addresses and ports, and can also contain several custom values in a Type-Length-Value format. More information about the Proxy Protocol can be found at https://www.haproxy.org/download/2.2/doc/proxy-protocol.txt
+
+From dnsdist to its backend
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+To enable the use of the proxy protocol between dnsdist and its backend, the ``useProxyProtocol`` parameter can be used when creating a :func:`new server <newServer>`:
```
Comment (chbruyand) [inline pdns/dnsdistdist/docs/advanced/passing-source-address.rst:26]:
```suggestion
To enable the use of the Proxy Protocol between dnsdist and its backend, the ``useProxyProtocol`` parameter can be used when creating a :func:`new server <newServer>`:
```

File: pdns/dnsdistdist/docs/advanced/passing-source-address.rst [hunk 2]
```diff
@@ -4,19 +4,94 @@ Passing the source address to the backend
...
+From clients to dnsdist
+^^^^^^^^^^^^^^^^^^^^^^^
+
+Such a Proxy Protocol header can also be passed from the client to dnsdist, using :func:`setProxyProtocolACL` to specify which clients to accept it from:
+
+.. code-block:: lua
+
+  setProxyProtocolACL({'192.0.2.0/24'})
+
+Note that a proxy protocol payload will be required from these clients, regular DNS queries will no longer be accepted if they are not preceded by a proxy protocol payload.
```
Comment (chbruyand) [inline pdns/dnsdistdist/docs/advanced/passing-source-address.rst:49]:
```suggestion
Note that a Proxy Protocol payload will be required from these clients, regular DNS queries will no longer be accepted if they are not preceded by a Proxy Protocol payload.
```

**Review by chbruyand — APPROVED** [review]

**Pull request merged**